fusion/ruby/sample.rb

```diff
@@ -0,0 +1,171 @@
+require "json"
+require "http"
+require "optparse"
+
+
+# Place holders for Yelp Fusion's OAuth 2.0 credentials. Grab them
+# from https://www.yelp.com/developers/v3/manage_app
+CLIENT_ID = nil
+CLIENT_SECRET = nil
+
+
+# Constants, do not change these
+API_HOST = "https://api.yelp.com"
+SEARCH_PATH = "/v3/businesses/search"
+BUSINESS_PATH = "/v3/businesses/"  # trailing / because we append the business id to the path
+TOKEN_PATH = "/oauth2/token"
+GRANT_TYPE = "client_credentials"
+
+
+DEFAULT_BUSINESS_ID = "yelp-san-francisco"
+DEFAULT_TERM = "dinner"
+DEFAULT_LOCATION = "San Francisco, CA"
+SEARCH_LIMIT = 5
+
+
+# Make a request to the Fusion API token endpoint to get the access token.
+# 
+# host - the API's host
+# path - the oauth2 token path
+#
+# Examples
+#
+#   bearer_token
+#   # => "Bearer some_fake_access_token"
+#
+# Returns your access token
+def bearer_token
+  # Put the url together
+  url = "#{API_HOST}#{TOKEN_PATH}"
+
+  raise "Please set your CLIENT_ID" if CLIENT_ID.nil?
+  raise "Please set your CLIENT_SECRET" if CLIENT_SECRET.nil?
+
+  # Build our params hash
+  params = {
+    client_id: CLIENT_ID,
+    client_secret: CLIENT_SECRET,
+    grant_type: GRANT_TYPE
+  }
+
+  response = HTTP.post(url, params: params)
+  parsed = response.parse
+
+  "#{parsed['token_type']} #{parsed['access_token']}"
+end
+
+
+# Make a request to the Fusion search endpoint. Full documentation is online at:
+# https://www.yelp.com/developers/documentation/v3/business_search
+#
+# term - search term used to find businesses
+# location - what geographic location the search should happen
+#
+# Examples
+#
+#   search("burrito", "san francisco")
+#   # => {
+#          "total": 1000000,
+#          "businesses": [
+#            "name": "El Farolito"
+#            ...
+#          ]
+#        }
+#
+#   search("sea food", "Seattle")
+#   # => {
+#          "total": 1432,
+#          "businesses": [
+#            "name": "Taylor Shellfish Farms"
+#            ...
+#          ]
+#        }
+#
+# Returns a parsed json object of the request
+def search(term, location)
+  url = "#{API_HOST}#{SEARCH_PATH}"
+  params = {
+    term: term,
+    location: location,
+    limit: SEARCH_LIMIT
+  }
+
+  response = HTTP.auth(bearer_token).get(url, params: params)
+  response.parse
+end
+
+
+# Look up a business by a given business id. Full documentation is online at:
+# https://www.yelp.com/developers/documentation/v3/business
+# 
+# business_id - a string business id
+#
+# Examples
+# 
+#   business("yelp-san-francisco")
+#   # => {
+#          "name": "Yelp",
+#          "id": "yelp-san-francisco"
+#          ...
+#        }
+#
+# Returns a parsed json object of the request
+def business(business_id)
+  url = "#{API_HOST}#{BUSINESS_PATH}#{business_id}"
```

Comment:
Do we need
`ERB::Util.url_encode(business_id)`
like in still-open pull request https://github.com/Yelp/yelp-ruby/pull/55 ?